compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -535,6 +550,7 @@ pub fn compile_declarative_macro(
         lhses,
         rhses,
         valid,
+        def_id,
```

Comment:
```suggestion
        def.id != DUMMY_NODE_ID
```
I'd reduce this `DefId` to a boolean `is_local` flag, since knowing the full `DefId` is not necessary here.
The value of that flag is already known here without additionally passing it to `compile_declarative_macro`.
See how `fn buffer_lint` in `compiler\rustc_expand\src\mbe\macro_check.rs` works.
(TODO: Replace all `NodeId`s with boolean `is_local` flags in `compiler\rustc_expand\src\mbe\macro_check.rs` too.)